streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1054,9 +1101,21 @@ private void suspendTasksAndState()  {
 
         final AtomicReference<RuntimeException> firstException = new AtomicReference<>(null);
 
-        for (final AbstractTask task : activeAndStandbytasks()) {
-            try {
+        firstException.compareAndSet(null, performOnStreamTasks(new StreamTaskAction() {
+            @Override
+            public String name() {
+                return "suspend";
+            }
+
+            @Override
+            public void apply(final StreamTask task) {
                 task.suspend();
+            }
+        }));
+
+        for (final StandbyTask task : standbyTasks.values()) {
```

Comment:
Why we want to explicitly commit the standby tasks here? Or put in other words, if we do need to commit after suspend for standby tasks as well, should we just call `commit` inside `suspend` as we did for StreamTask as well?